robot/basestation/static/js/elements/navigation/localization.js

```diff
@@ -0,0 +1,80 @@
+let rotationValue = 0;
+let zoomValue = 1;
+let x = 0
+let y = 0;
+let headingElt;
+
+function setup(){
+    let canvas = createCanvas(450,300);
+    canvas.parent('#canvasContainer');
+    angleMode(DEGREES);
+}
+
+
+function draw() {
+    // keycodes
+    // q = 81, e = 69, r = 82, w = 87, a = 65, s = 83, d = 68
```

Comment:
could you use constants. No magic values are accepted.

Reply:
i made a JSON with the keycodes so i call the letter instead of its value.

Reply:
This kind of stuff is usually handled by the library. I would be surprised if `p5` doesn't supply you constants of the type Key.E_KEY

Reply:
so I looked it up, it appears to be the case that they have NOT declared these types of constants lol. They have them for UP,LEFT, RIGHT, etc, but not for like W or U AFAIK

Reply:
furthering my search. To test out keys ; http://keycode.info/

Also check this out : https://stackoverflow.com/a/1465409

Reply:
yes i used keycode.info to get the keycodes for the letters